File: app/src/main/java/com/artmurka/artmurkaapp/Presenter/OrdersPresenter.java
package com.artmurka.artmurkaapp.Presenter;

import android.util.Log;

import com.artmurka.artmurkaapp.Model.InterfacesModel.IOrderRequest;
import com.artmurka.artmurkaapp.Model.Modules.OrdersRequests;
import com.artmurka.artmurkaapp.Model.Pojo.ItemList.Orders.Orders;
import com.artmurka.artmurkaapp.Presenter.InterfacesPresenter.IOrderPresenter;
import com.artmurka.artmurkaapp.Views.Fragments.Interfaces.IOrderFragment;
import com.google.gson.Gson;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class OrdersPresenter implements IOrderPresenter{
    IOrderFragment fragment;

    public OrdersPresenter(IOrderFragment fragment){
        this.fragment = fragment;

    }

    @Override
    public void getOrders() {
        IOrderRequest request = new OrdersRequests();
        Call<Orders> orders = request.getOrders();
        orders.enqueue(new Callback<Orders>() {
            @Override
            public void onResponse(Call<Orders> call, Response<Orders> response) {
                fragment.showOrders(response.body());
            }

            @Override
            public void onFailure(Call<Orders> call, Throwable t) {

            }
        });
    }
}

File: app/src/main/java/com/artmurka/artmurkaapp/presenter/OrdersPresenter.kt
package com.artmurka.artmurkaapp.presenter

import com.artmurka.artmurkaapp.data.model.pojo.itemlist.orders.Orders
import com.artmurka.artmurkaapp.presenter.interfaces_presenter.IOrderPresenter
import com.artmurka.artmurkaapp.android.views.fragments.interfaces.IOrderFragment
import com.artmurka.artmurkaapp.domain.usecase.orders.GetOrdersUseCase
import io.reactivex.observers.DisposableObserver

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import javax.inject.Inject

class OrdersPresenter @Inject constructor(val request: GetOrdersUseCase) : BasePresenter<IOrderFragment>(), IOrderPresenter {

    override fun getOrders() {
        request.execute(object : DisposableObserver<Orders>() {
            override fun onComplete() {}

            override fun onNext(t: Orders) {
                view?.showOrders(t)
            }

            override fun onError(e: Throwable) {}
        }, GetOrdersUseCase.Params())
    }

    override fun onDropView() {
        request.dispose()
        super.onDropView()
    }
}
